modules/core/src/main/java/org/apache/synapse/message/processor/impl/forwarder/ForwardingService.java

```diff
@@ -262,80 +285,99 @@ public void execute() {
+			 * we need to skip it here. Otherwise the cron expression is kept
+			 * sleeping twice as the forwarding interval.
+			 */
+			if (interval > 0 && interval < MessageProcessorConstants.THRESHOULD_INTERVAL &&
+					!isRunningUnderCronExpression()) {
+				try {
+					Thread.sleep(interval);
+				} catch (InterruptedException e) {
+					log.debug("Current Thread was interrupted while it is sleeping.");
+				}
+			}
+
+			/*
+			 * Gives the control back to Quartz scheduler. This needs to be done
+			 * only if the interval value is less than the Threshould interval
+			 * value of 1000 ms, where the scheduling is done outside of Quartz
+			 * via the while loop. Otherwise the schedular will get blocked.
+			 * For cron expressions with interval < 1000ms this scenario is not
+			 * applicable hence skipping it here. For cron expressions, all the
+			 * messages in the queue at the moment are sent to the backend. If
+			 * you give control back to the Quartz that behavior can not be
+			 * achieved, only a portion of the messages will get dispatched
+			 * while other messages will remain in the queue.
+			 */
+			if (isThrottling && new Date().getTime() - startTime > 1000 &&
+					!isRunningUnderCronExpression()) {
+				break;
+			}
 		} while ((isThrottling || isRunningUnderCronExpression()) && !isTerminated);
 
 		if (log.isDebugEnabled()) {
 			log.debug("Exiting service thread of message processor [" +
-			          this.messageProcessor.getName() + "]");
-		}
-	}
-
-    /**
-     * Helper method to get a value of a parameters in the AxisConfiguration
-     *
-     * @param axisConfiguration AxisConfiguration instance
-     * @param paramKey The name / key of the parameter
-     * @return The value of the parameter
-     */
-    private static String getAxis2ParameterValue(AxisConfiguration axisConfiguration,
-                                                 String paramKey) {
-        Parameter parameter = axisConfiguration.getParameter(paramKey);
-        if (parameter == null) {
-            return null;
-        }
-        Object value = parameter.getValue();
-        if (value != null && value instanceof String) {
-            return (String) parameter.getValue();
-        } else {
-            return null;
-        }
-    }
-
-	public void init(SynapseEnvironment se) {
+					this.messageProcessor.getName() + "]");
+		}
+	}
+
+	/**
+	 * Helper method to get a value of a parameters in the AxisConfiguration
+	 *
+	 * @param axisConfiguration AxisConfiguration instance
+	 * @param paramKey          The name / key of the parameter
+	 * @return The value of the parameter
+	 */
+	private static String getAxis2ParameterValue(AxisConfiguration axisConfiguration,
+												 String paramKey) {
+		Parameter parameter = axisConfiguration.getParameter(paramKey);
+		if (parameter == null) {
+			return null;
+		}
+		Object value = parameter.getValue();
+		if (value != null && value instanceof String) {
+			return (String) parameter.getValue();
```

Comment:
We can return the value here rather than getting parameter.getValue() ?

Reply:
Removed method as of no usages